sdk/python/feast/registry.py

```diff
@@ -35,26 +36,33 @@
 
 REGISTRY_SCHEMA_VERSION = "1"
 
+warnings.simplefilter("once", DeprecationWarning)
+
 
 class Registry:
     """
     Registry: A registry allows for the management and persistence of feature definitions and related metadata.
     """
 
-    cached_registry_proto: Optional[RegistryProto] = None
-    cached_registry_proto_created: Optional[datetime] = None
-    cached_registry_proto_ttl: timedelta
-    cache_being_updated: bool = False
+    cached_registry_proto: RegistryProto
 
-    def __init__(self, registry_path: str, repo_path: Path, cache_ttl: timedelta):
+    def __init__(
+        self,
+        registry_path: str,
+        repo_path: Path,
+        cache_ttl: timedelta = None,
+        instantiated_internally=False,
+    ):
         """
-        Create the Registry object.
+        Create the Registry object. It uses a write-back cache for speed, so the cached
+        registry proto is always up-to-date.
 
         Args:
             repo_path: Path to the base of the Feast repository
-            cache_ttl: The amount of time that cached registry state stays valid
+            cache_ttl: [Deprecated] Unnecessary since the cache is always up-to-date
             registry_path: filepath or GCS URI that is the location of the object store registry,
             or where it will be created if it does not exist yet.
+            instantiated_internally: Whether this Registry is being created and used by Feast
```

Comment:
`instantiated_internally` is a bit of an obscure name. What if we just support kwargs and then set a Feast specific key? otherwise we may be confusing users who don't know how to configure this argument.

Reply:
Good point; I agree that using kwargs should ensure that users don't get confused.